src/audio-manager.ts

```diff
@@ -1,261 +1,389 @@
-import {_audioContext} from './audio-listener.js';
+import {_audioContext, unlockAudioContext} from './audio-listener.js';
+import {Emitter} from '@wonderlandengine/api';
+import {
+    BufferPlayer,
+    DEF_VOL,
+    MIN_RAMP_TIME,
+    MIN_VOLUME,
+    OneShotPlayer,
+} from './audio-players.js';
 
-const RAMP_TIME = 20 / 1000;
+/**
+ * Enumerates the available channels within the AudioManager.
+ * These channels can be utilized to regulate the global volume of audio.
+ */
+export enum Channel {
+    /** Intended for sound effects. Connects to Master Channel. */
+    SFX,
+    /** Intended for music. Connects to Master Channel. */
+    MUSIC,
+    /** Connects directly to output. */
+    MASTER,
+}
+
+/**
+ * Enumerates the possible states of playback for audio sources.
+ */
+export enum PlayState {
+    /** The source has loaded and is ready to be played */
+    READY,
+    /** The source has started playing */
+    PLAYING,
+    /** The source has stopped */
+    STOPPED,
+}
+
+/**
+ * Represents a combination of a unique identifier and a play state.
+ */
+type PlayStateWithID = {
+    /** Unique identifier associated with the audio source. */
+    id: number;
+    /** Current state of playback for the audio source. */
+    state: PlayState;
+};
+
+/**
+ * Combines all settings for configuring playback in the AudioManager.
+ *
+ * @note The playOneShot() function utilizes this configuration for consistent playback settings. However, one-shots
+ * are incapable of looping and do not alter their audio channel.
+ */
+export type PlayConfig = {
+    /** Sets the volume of the player (0-1) */
+    volume?: number;
+    /** Whether to loop the audio */
+    loop?: boolean;
+    /**
+     * Sets the position of the audio source and makes it spatial.
+     *
+     * @note Panned audio will always use HRTF for spatialization.
+     */
+    position?: Float32Array;
+    /** Sets the channel on which the audio will be played */
+    channel?: Channel;
+};
+
+/**
+ * Default number of one-shot players.
+ * @todo: Question for Timmy: From your experience, how many one-shots/regular stuff (loops, music, etc) do you
+ * typically
+ * need?
+ */
+export const DEF_ONESHT_PLR_COUNT = 16;
```

Comment:
I use 25 channels for the audio in my games, but that number is not based on anything. I don't think I ever went over it. Even 16 sounds playing at the same time is a lot. If for some reason someone plays 16 at the same time and the 17th starts, stopping the first it's probably such a cacophony of noise no one will notice.  I think 16 is enough. 

And if for some odd reason sound plays such an important role in your game someone might need more for their game, they probably would use something like FMOD or build something themselves.